README.md

```diff
@@ -27,51 +29,57 @@ You can use the prebuilt binaries at the [releases page](https://github.com/Syma
+The `keymasterd` service runs the following services:
+* **Service Web Interface (default port 443)**: Access to the web interface running on port 443 (default) can be granted via LDAP or apache username/password files. For password backend Keymaster supports LDAP backends and apache password files.
+* **Admin Management Interface (default port 6920)**: The service exposed on port 6920 allows administrators or log collection systems to collect logs generated by the `keymasterd` service.
+
+To run `keymasterd` you will need to generate a config file. `keymasterd` facilitates this through the command-line arguments `-generateConfig` and `-alsoLogToStderr`. Running the `keymasterd` binary with these arguments will generate the following:
+* A configuration file. By default `keymasterd` will write this file to `/etc/keymaster/config.yml`.
+* The Keymaster CA key pair. The encrypted private key  (`masterkey.asc`) is an armored PGP file. For development (or if your trust model permits it) you can decrypt the private-key and write it to the filesystem. To decrypt the key run `gpg -ad $Filename`. Once decrypted set the `ssh_ca_filename` field in the `keymasterd` config file to the path of the decrypted master key.
+* Server keys (for Testing Purposes only): the `server.pem` and `server.key` files are generated to ????
+* Admin CA certificate and key: The admin CA certificate (`adminCA.pem`) and key (`adminCA.key`) are used to generate certificates that grant access to the control port of the `keymasterd` management interface (default port 443).
+
+Notice: Keymaster has a bug where the directory locations are not written correctly to the config file. Depending on the platform you're running Keymaster on the following workaround will apply:
+* RPM (CentOS): Modify the following configuration items in your `config.yml` file:
+    * `data_directory: /var/lib/keymaster `
+    * `shared_data_directory: /usr/share/keymasterd/`.
+* DEB (Debian/Ubuntu): Modify the following configuration items in your `config.yml` file:
+    * `data_directory: /var/lib/keymaster `
+    * `shared_data_directory: /usr/share/keymasterd/`.
+
+##### Supported backend authentication methods
+Several authentication methods are supported by the `keymasterd` service. You can separately specify which authentication methods you accept for the web backend (`allowed_auth_backends_for_webui`) and for obtaining certificates (`allowed_auth_backends_for_certs`).
+* **LDAP**: For LDAP the `bind_pattern` is a printf string where `%s` is the place where the username will be substituted. For example for an 389ds/openldap string might be: `"uid=%s,ou=People,dc=example,dc=com`. To leverage LDAP authentication set the appropriate `allowed_auth_*` setting to `["federated"]`.
```

Comment:
The setting for ldap is password. ["federated"] is for using keymaster web-ui to be behind an oauth2 provider such as github or google.